src/main/java/com/minecolonies/entity/ai/citizen/lumberjack/Tree.java

```diff
@@ -193,13 +208,66 @@ public static boolean checkTree(@NotNull IBlockAccess world, BlockPos pos)
             return false;
         }
 
+        BlockPos[] baseAndTOp = getBottomAndTopLog(world, pos, new LinkedList<>(), null, null);
+
+        if(baseAndTOp == null || baseAndTOp.length < 2)
+        {
+            return false;
+        }
+
         //Get base log, should already be base log
-        BlockPos basePos = getBaseLog(world, pos);
+        BlockPos basePos = baseAndTOp[0];
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'basePos' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)